File: app/src/main/java/com/saizad/mvvmexample/components/main/home/UserList.java
package com.saizad.mvvmexample.components.main.home;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

import com.saizad.mvvmexample.R;
import com.saizad.mvvmexample.models.User;

import sa.zad.pagedrecyclerlist.IntegerPageKeyList;


public class UserList extends IntegerPageKeyList<User, UserListItem> {
    public UserList(Context context) {
        this(context, null);
    }

    public UserList(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public UserList(Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public UserListItem getSelectorItem(Context context, int viewType) {
        return (UserListItem) View.inflate(context, R.layout.item_user_list, null);
    }
}

File: app/src/main/java/com/saizad/mvvmexample/components/main/home/UserList.kt
package com.saizad.mvvmexample.components.main.home

import android.content.Context
import android.util.AttributeSet
import android.view.View


import androidx.annotation.Nullable
import com.saizad.mvvmexample.R
import sa.zad.pagedrecyclerlist.IntegerPageKeyList
import com.saizad.mvvmexample.models.User


class UserList @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null, defStyle: Int = 0
) : IntegerPageKeyList<User, UserItem>(context, attrs, defStyle) {

    override fun getSelectorItem(context: Context, viewType: Int): UserItem {
        return View.inflate(context, R.layout.item_user, null) as UserItem
    }
}